### PR TITLE
one logger per module

### LINKED ISSUE #29: one logger per module

### PR DESCRIPTION
**Pull request by julia326**
fixes #29

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/datacache/30)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -61,6 +61,7 @@
             "biopython>=1.65",
             "requests>=2.5.1",
             "typechecks>=0.0.2",
+            "mock",
```
Comment (ihodes) [inline setup.py]: was this an implicit requirement?

File: datacache/database.py [hunk 2]
```diff
@@ -17,6 +17,10 @@
 
 from typechecks import require_integer, require_string, require_iterable_of
 
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
```
Comment (iskandr) [inline datacache/database.py]: How do we externally configure the verbosity of logging output? I sometimes want to run a script with maximal debug logging but in other cases (e.g. Isaac sifting through 10mb of log messages) we might only want WARN.

File: setup.py (hunk 1 above)
Comment (julia326) [inline setup.py]: yeah, it was - failed when running 'nosetests' so i figured best to add

File: datacache/database.py (hunk 2 above)
Comment (julia326) [inline datacache/database.py]: will address this here, but it applies to all the other logger PRs you guys will see.

tl;dr: use config files for each binary in the stack, add --verbose flag to each binary that'll set the console handler output to DEBUG (otherwise it'll default to INFO or WARN or something)

long version of the answer:
- do absolutely no log config in module code
- have all configs in a file (<package>/logging.conf) that’s read from the entry point to any of the binaries, and each one can set all the loggers/handlers under it as necessary
- idea being: let’s say i don’t care in the vaxrank binary that mhctools is making a bunch of netmhc calls. so then in the vaxrank config, i can set the netmhc *logger* to level INFO (rather than debug). but i care about e.g. varcode debug statements, so i set the varcode logger to level DEBUG
- say i’m in verbose mode, but i still don’t give a shit about mhctools debug calls. then mhctools logger level will be INFO (to suppress debug statements from that logger), but the console handler will be DEBUG (to see other debug statements from various other places)
